### ISSUE TITLE
Rename capitalized directory to lowercase fails

### ISSUE THREAD

**Issue by smackesey** [ranger/ranger#1528]
#### Runtime Environment

- Operating system and version: macOS 10.14.4
- Terminal emulator and version: iTerm2 3.2.4 beta3
- Python version: 2.7.10
- Ranger version/commit: 1.9.2
- Locale: None,None

#### Current Behavior

Attempting to rename a capitalized directory name to lowercase variant fails with error message. For example, create a directory `Testdir`; try `rename testdir` when you have `Testdir` highlighted. It fails with message:

    Can't rename; file already exists!

#### Expected Behavior

It should change the name to the lowercase variant.

#### Steps to reproduce
1. create directory `Testdir`
2. try `rename testdir` on highlighted `Testdir`
3. observe error

**Comment by toonn** [comment]
The problem is probably you're on a case-insensitive file system. Maybe we can add a check whether the name clash is the same file? Though that'll mean redundant moves/renames if you don't change the name.

**Comment by smackesey** [comment]
Yes-- never knew that the macOS filesystem (APFS) was case-insensitive by default, but apparently it is. I don't know ranger's internals so I don't know how best to address the problem, but it seems clear that this is a bug since `Testdir` and `testdir` remain different values in many contexts  (e.g. display), if not for the purposes of file comparison. The solution you propose sounds good-- the redundant moves/renames problem could be avoided by doing a Python-based (i.e. case-sensitive) comparison of the existing filename and the supplied new name.

**Comment by toonn** [comment]
Well not quite. You need to ask the filesystem "Does *testdir* exist?" and on a case-insensitive filesystem it'll say yes if a file named "Testdir" exists. So it's an *extra* check. That's 2x the effort for *all* rename/move/copy operations just because *some* silly filesystems are case-insensitive : )

Temporarily you should be able to work around this by using `:bulkrename` or just `:shell mv %s newname.txt`.
